src/OpenLoco/Objects/ObjectManager.cpp

```diff
@@ -569,6 +574,34 @@ namespace OpenLoco::ObjectManager
         return checksum == object.checksum;
     }
 
+    static bool partialLoad(const ObjectHeader& header, stdx::span<uint8_t> objectData)
+    {
+        auto type = header.getType();
+        size_t index = 0;
+        for (; index < getMaxObjects(type); ++index)
+        {
+            if (object_repository[static_cast<uint8_t>(type)].objects[index] == reinterpret_cast<object*>(-1))
+            {
+                break;
+            }
+        }
+        // No slot found. Not possible except if unload all had not been called
+        if (index >= getMaxObjects(type))
+        {
+            return false;
+        }
+
+        auto* obj = reinterpret_cast<object*>(objectData.data());
+        object_repository[static_cast<uint8_t>(type)].objects[index] = obj;
+        object_repository[static_cast<uint8_t>(type)].object_entry_extendeds[index] = ObjectEntry2(header, objectData.size());
+        return true;
+    }
+
+    // 0x0047285C
+    bool installObject(const ObjectHeader& objectHeader)
+    {
+    }
```

Comment:
Does this still need to call vanilla in the mean time?

Reply:
Must be fully implemented unfortunately